### PR TITLE
Flying Things Decoder

### PR DESCRIPTION
**Pull request by phillip-thomas-pd**

### CODE REVIEW

File: paralleldomain/decoding/flying_things/sensor_frame_decoder.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    def _decode_class_maps(self) -> Dict[AnnotationType, ClassMap]:
+        return dict()
+
+    def _decode_available_annotation_types(
+        self, sensor_name: SensorName, frame_id: FrameId
+    ) -> Dict[AnnotationType, AnnotationIdentifier]:
+        optical_flow_file_name = f"{sensor_name}/{frame_id}"
+
+        return {
```
Comment (pd-michaelstanley) [inline paralleldomain/decoding/flying_things/sensor_frame_decoder.py:102]: Images at the beg/end of scenes don't have both flow types (e.g. 1st frame is missing into_past flow). You handle missing flow files in _decode_optical_flow() lower down, but should we indicate here that there is not an OpticalFlow annotation available? I'm not sure of the convention.

File: paralleldomain/decoding/flying_things/common.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        data = np.fromfile(file, endian + "f")
+        shape = (height, width, 3) if color else (height, width)
+
+        data = np.reshape(data, shape)
+        data = np.flipud(data)
+        return data, scale
+
+
+def read_flow(file_path: AnyPath) -> np.ndarray:
```
Comment (pd-michaelstanley) [inline paralleldomain/decoding/flying_things/common.py:138]: No big deal, but this is only used in sensor_frame_decoder.py so probably should live there.

File: paralleldomain/decoding/flying_things/common.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            / sensor_name
+        )
+        frame_ids = {img.split(".png")[0] for img in folder_path.glob("*.png")}
+        return frame_ids
+    else:
+        frame_ids = get_frame_ids_of_subset_scene(scene_name=scene_name, split_list=split_list)
+        return set(frame_ids)
+
+
+def read_pfm(file_path: AnyPath) -> Tuple[np.ndarray, float]:
```
Comment (pd-michaelstanley) [inline paralleldomain/decoding/flying_things/common.py:104]: No big deal, but this is only used in sensor_frame_decoder.py so probably should live there.

File: paralleldomain/decoding/kitti_flow/decoder.py [hunk 4]
```diff
@@ -160,7 +160,7 @@ def _create_camera_sensor_decoder(
     def _create_lidar_sensor_decoder(
         self, scene_name: SceneName, lidar_name: SensorName, dataset_name: str
     ) -> LidarSensorDecoder[datetime]:
-        raise ValueError("Directory decoder does not support lidar data!")
+        raise ValueError("KITTI-flow does not support lidar data!")
```
Comment (pd-michaelstanley) [inline paralleldomain/decoding/kitti_flow/decoder.py:163]: I cleaned up these error moessagess in kitti-flow and flyingchairs as well, that's why those files are in this PR now.

File: paralleldomain/decoding/flying_things/sensor_frame_decoder.py (hunk 1 above)
Comment (phillip-thomas-pd) [reply paralleldomain/decoding/flying_things/sensor_frame_decoder.py:100]: The way i understood this format, there is always either a forward or a backward flow available for a frame. At the start or end of a sequence one of them would be none then. But the OpticalFlow annotation type would be available on every frame.

**Pull request merged**